### PR TITLE
Add booth types to booths

### PR DESCRIPTION
**Pull request by rishabhptr**
In certain cases we need a type field for stands which could be dropdown from which the user can choose.
In case of FOSDEM it could be 'one table' or 'two table', these may not be the same for all conferences and may even change from year to year therefore it should be customisable.

Screenshots:
Screenshot from 2019-08-17 18-21-04
Screenshot from 2019-08-17 18-21-11
Screenshot from 2019-08-17 18-21-31
Screenshot from 2019-08-17 18-23-46

### CODE REVIEW

File: app/views/booths/_form.html.haml [hunk 1]
```diff
@@ -10,6 +10,14 @@
         = f.input :submitter_relationship, input_html: { rows: 5, data: { provide: 'markdown-editable' } }, required: true,
           label: 'Submitter\'s relation',
           hint: 'e.g. employee, comunity manager, etc'
+
+        = f.input :booth_type_id, as: :select,
+          collection: conference.program.booth_types.map {|type| ["#{type.title}", type.id]},
+          include_blank: false, label: 'Type', input_html: { class: 'select-help-toggle' } if conference.program.booth_types.any?
```
Comment (differentreality) [inline app/views/booths/_form.html.haml]: Perhaps it doesn't always make sense to enforce the selection of the type. I'd suggest we do include the blank option with a relevant message (Please select, or sth similar to our other forms)

File: app/views/admin/booths/new.html.haml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 %h1 New #{t'booth'}
 
-= render 'booths/form'
+= render 'booths/form', conference: @conference
```
Comment (differentreality) [inline app/views/admin/booths/new.html.haml]: Why do we need to pass the conference variable and not use @conference inside the form?

File: app/views/booths/_form.html.haml [hunk 3]
```diff
@@ -10,6 +10,14 @@
...
+
+        - conference.program.booth_types.each do |booth_type|
```
Comment (differentreality) [inline app/views/booths/_form.html.haml]: Can you show me what you expect to show with this? Perhaps it makes sense for the type description to be a hint under the input field?

File: app/views/admin/booths/new.html.haml (hunk 2 above)
Comment (rishabhptr) [reply app/views/admin/booths/new.html.haml]: @differentreality Actually [haml-lint](https://github.com/sds/haml-lint) keeps complaining about using instance variable in partials. This has been issue in my other pr's too, maybe I should go ahead and generate a to-do file for all these changes?

File: app/views/booths/_form.html.haml (hunk 1 above)
Comment (rishabhptr) [reply app/views/booths/_form.html.haml]: @differentreality sure :+1:

File: app/views/booths/_form.html.haml (hunk 3 above)
Comment (rishabhptr) [reply app/views/booths/_form.html.haml]: That is what happens currently, whatever type is chosen its description appears as a hint below the field .
image

**Comment by differentreality** [comment]
Looking at this again with fresher eyes, I am wondering if it makes more sense for this feature to be integrated as a survey. 

We don't currently have an option to show a survey while creating a booth though.

**Pull request closed without merging**